Extensions/TopDownMovementBehavior/topdownmovementruntimebehavior.ts

```diff
@@ -366,8 +391,8 @@ namespace gdjs {
       const squaredSpeed =
         this._xVelocity * this._xVelocity + this._yVelocity * this._yVelocity;
       if (squaredSpeed > this._maxSpeed * this._maxSpeed) {
-        this._xVelocity = this._maxSpeed * Math.cos(directionInRad);
-        this._yVelocity = this._maxSpeed * Math.sin(directionInRad);
+        this._xVelocity = this._maxSpeed * cos;
```

Comment:
Is there a risk cos and sin are 0 here?

Reply:
`cos` and `sin` are calculated whenever `directionInRad` is set. So `directionInRad` would be `0` too.
It happens when there is no input and the object is stopped. This code is used when the maximum speed is reached.

It's not very clean, but the optimization is interesting.

Reply:
I though about it too but It would bring more confusion. There is no default direction the object will take. `0` means not yet initialized here.

Reply:
Ok, as you prefer but because 0 is a valid value I'm unsure if this will create confusion if someone add a new branch or rework the code order. Feel free to merge though!